internal/v2/tls_config_store/tls_config_store.go

```diff
@@ -37,7 +41,7 @@ func GetTlsConfigurationForClient() *tls.Config {
 	// Create mTLS credentials for client.
 	return &tls.Config {
 		Certificates: []tls.Certificate{cert},
-		VerifyPeerCertificate: verifyPeerCertificateFunc("s2a_test_cert", rootCertPool), // TODO(rmehta19): Call cert verifier library.
+		VerifyPeerCertificate: certverifier.VerifyClientCertificateChain("s2a_test_cert", rootCertPool),
```

Comment:
When building the tls.Config for the client, we should use the VerifyServerCertificateChain function (because the client will be verifying the server's certificate).

This also means we need to pass in the hostname to that API. We can add a "serverHostname" string parameter to the GetTlsConfigurationForClient function. When we call GetTlsConfigurationForClient from ClientHandshake(), we can pass the serverAuthority parameter to it.

Reply:
Thanks for catching this Matt, will do!